x-pack/plugins/enterprise_search/server/lib/indices/create_api_key.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { KibanaRequest } from '@kbn/core/server';
+import { SecurityPluginStart } from '@kbn/security-plugin/server';
+
+export const createApiKey = async (
+  request: KibanaRequest,
+  security: SecurityPluginStart,
+  indexName: string,
+  keyName: string
+) => {
+  return await security.authc.apiKeys.create(request, {
+    name: keyName,
+    role_descriptors: {
+      [`${indexName}-key-role`]: {
+        cluster: [],
+        index: [
+          {
+            names: [indexName],
+            privileges: ['all'],
```

Comment:
Is it possible to hit this endpoint and get an API key with "all" privileges as a user that shouldn't access that power. I am trying to ask if we are creating an issue by not using `core.elasticsearch.client.asCurrentUser.something`

Reply:
The method provided by the security plugin is ultimately run as the current user...that is why it takes a `KibanaRequest` as one of its parameters.